test/protocol/data/redis/check_redis.c

```diff
@@ -0,0 +1,540 @@
+#include <protocol/data/redis_include.h>
+
+#include <buffer/cc_buf.h>
+#include <cc_array.h>
+#include <cc_bstring.h>
+#include <cc_define.h>
+
+#include <check.h>
+
+#include <stdio.h>
+#include <stdlib.h>
+
+/* define for each suite, local scope due to macro visibility rule */
+#define SUITE_NAME "redis"
+#define DEBUG_LOG  SUITE_NAME ".log"
+
+struct request *req;
+struct response *rsp;
+struct buf *buf;
+
+/*
+ * utilities
+ */
+static void
+test_setup(void)
+{
+    req = request_create();
+    rsp = response_create();
+    buf = buf_create();
+}
+
+static void
+test_reset(void)
+{
+    request_reset(req);
+    response_reset(rsp);
+    buf_reset(buf);
+}
+
+static void
+test_teardown(void)
+{
+    buf_destroy(&buf);
+    response_destroy(&rsp);
+    request_destroy(&req);
+}
+
+/**************
+ * test cases *
+ **************/
+
+/*
+ * token
+ */
+START_TEST(test_simple_string)
+{
+#define SERIALIZED "+foobar\r\n"
+#define STR "foobar"
```

Comment:
I thought about it before, it is less error-prone if we change test cases, on the other hand it is less readable by themselves. You are probably still right though it end up a little better.